plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -257,17 +257,10 @@ public List<SchemaTableName> listTables(ConnectorSession session, Optional<Strin
 
     private List<SchemaTableName> listTablesInRemoteSchema(BigQueryClient client, String projectId, String remoteSchemaName)
     {
-        ImmutableList.Builder<SchemaTableName> tableNames = ImmutableList.builder();
+        List<SchemaTableName> tableNames = ImmutableList.of();
         try {
             Iterable<TableId> tableIds = client.listTableIds(DatasetId.of(projectId, remoteSchemaName));
-            for (TableId tableId : tableIds) {
-                // filter ambiguous tables
-                client.toRemoteTable(projectId, remoteSchemaName, tableId.getTable().toLowerCase(ENGLISH), tableIds)
-                        .filter(RemoteDatabaseObject::isAmbiguous)
-                        .ifPresentOrElse(
-                                remoteTable -> log.debug("Filtered out [%s.%s] from list of tables due to ambiguous name", remoteSchemaName, tableId.getTable()),
-                                () -> tableNames.add(new SchemaTableName(client.toSchemaName(DatasetId.of(projectId, tableId.getDataset())), tableId.getTable())));
-            }
+            tableNames = client.getNonAmbiguousSchemaTableNames(projectId, remoteSchemaName, tableIds);
```

Comment:
tableNames variable is redundant:
```suggestion
            return client.getNonAmbiguousSchemaTableNames(projectId, remoteSchemaName, tableIds);
```

Reply:
its not since we return empty list in catch. I can get rid of it and return empty there instead.